torch/utils/weak.py

```diff
@@ -83,7 +82,7 @@ def __eq__(self, other):
 
 # This is directly adapted from cpython/Lib/weakref.py
 class WeakIdKeyDictionary(MutableMapping):
-    data: Dict[WeakIdRef, object]
+    data: dict[WeakIdRef, object]
```

Comment:
`dict` is not subscriptable in 3.8, is it?
```
% python
Python 3.8.16 (default, Mar  1 2023, 21:18:45) 
[Clang 14.0.6 ] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> dict[int, str]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'type' object is not subscriptable
```

Reply:
This is ok because of the first line:

`from __future__ import annotations`

and because this is an annotation, not an expression. aka. it is not evaluated by Python.

Reply:
Unless we do runtime type checking somewhere in here

Reply:
Yeah, I double checked, it's OK because of the future annotation.

Reply:
LGTM then

Reply:
@ezyang though I wonder if torch.jit frontend will recognize this one (I vaguely remember adding support for subscriptable `dict` to JIT frontend, but perhaps worth small test)

Reply:
It's a good question. In this particular case, it's no problem (because you're never going to TS this code), but something to keep an eye on elsewhere.